external/web3-blockchain-service.js

```diff
@@ -11,10 +11,48 @@ class Web3BlockchainService {
         this.gasStationLink = 'https://gasstation-mumbai.matic.today/v2';
     }
 
-    initialize(logger) {
+    async initialize(logger) {
         this.logger = logger;
         this.rpcNumber = 0;
-        this.web3 = new Web3(this.config.rpcEndpoints[0]);
+        await this.initializeWeb3();
+        await this.initializeContracts();
+    }
+
+    async initializeWeb3() {
+        let tries = 0;
+        let isRpcConnected = false;
+        while (!isRpcConnected) {
+            if (tries >= this.config.rpcEndpoints.length) {
+                throw Error('Blockchain initialisation failed');
```

Comment:
So when we throw an error here, will it be caught on the top and restart the node or it will be caught somewhere in between? 

Not sure if this is good place to discuss, but what if node enters auto-restarting loop, let's think if we should add some pause before every restart is executed, also to check the order of initialization steps so first check the modules that have higher chance to fail initialization.

Reply:
The error is caught both if it is thrown on initialisation of the service and if it is thrown while the node is running. The node should not get in a restart loop because of this.